packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -723,6 +723,8 @@ export class Function extends FunctionBase {
       this.validateProfiling(props);
       props.profilingGroup.grantPublish(this.role);
       profilingGroupEnvironmentVariables = {
+        AWS_CODEGURU_PROFILER_GROUP_NAME: props.profilingGroup.profilingGroupName,
+        AWS_CODEGURU_PROFILER_TARGET_REGION: Stack.of(scope).region,
```

Comment:
Would it make sense to get this as `profilingGroup.env.region`? The profiling group may not be in the same region as the Lambda. See https://docs.aws.amazon.com/codeguru/latest/profiler-ug/working-with-unsupported-regions.html

Reply:
thanks for the call out. I actually found that this wasn't correctly configured for profiling groups imported from arns. I've made a PR with a fix in #23568